File: advent_of_code/2025/day10_python/main_test.py
from main import (
    extract_data,
    solve_part1,
    solve_part2,
)


def test_extract_data():
    filename = "./data/input_test.txt"
    expected_input = [
        (".##.", [[3], [1, 3], [2], [2, 3], [0, 2], [0, 1]], (3, 5, 4, 7)),
        (
            "...#.",
            [[0, 2, 3, 4], [2, 3], [0, 4], [0, 1, 2], [1, 2, 3, 4]],
            (7, 5, 12, 7, 2),
        ),
        (
            ".###.#",
            [[0, 1, 2, 3, 4], [0, 3, 4], [0, 1, 2, 4, 5], [1, 2]],
            (10, 11, 11, 5, 10, 5),
        ),
    ]
    input = extract_data(filename)
    print("input:", input)
    assert input == expected_input


def test_solve_part1():
    filename = "./data/input_test.txt"
    assert solve_part1(filename) == 7


def test_solve_part2():
    filename = "./data/input_test.txt"
    assert solve_part2(filename) == 33

File: advent_of_code/2025/day07_ts_with_deno/main_test.ts
import { assertEquals } from "@std/assert";
import {
  Coordinate,
  displayGrid,
  findStart,
  Grid,
  parseInput,
  solvePart1,
  solvePart2,
} from "./main.ts";

Deno.test("Testing the parsing function on the test input data", function parseInputTest() {
  const actual: Grid = parseInput("./data/input_test.txt");
  assertEquals(actual.length, 16);
  assertEquals(actual[0].length, 15);
  displayGrid(actual);
});

Deno.test("Testing the function to find the start position", function findStartTest() {
  const expected: Coordinate = { i: 0, j: 7 };
  assertEquals(findStart(parseInput("./data/input_test.txt")), expected);
});

Deno.test("Solving part 1 on the test input data", function solvePart1Test() {
  const expected = 21;
  const actual = solvePart1("./data/input_test.txt");
  assertEquals(actual, expected);
});

Deno.test("Solving part 2 on the test input data", function solvePart1Test() {
  const expected = 40;
  const actual = solvePart2("./data/input_test.txt");
  assertEquals(actual, expected);
});
